axum/src/routing/tests/mod.rs

```diff
@@ -924,6 +925,26 @@ async fn state_isnt_cloned_too_much() {
     assert_eq!(state.count(), 3);
 }
 
+#[crate::test]
+async fn state_isnt_cloned_too_much_in_layer() {
```

Comment:
I'm not sure if it's the right place to put this test

Reply:
Fine by me.